testsuite/test_shell.py

```diff
@@ -7,6 +7,26 @@
 from testsuite.support import ROOT_DIR, PseudoFile
 
 
+def safe_line_split(line):
+    """ parses the path, from message details """
+
+    # On certain platforms, the drive notation conflicts
+    # with the message seperator, inducing problems during tuple unpacking.
+
+    split = line.split(':')
+    if 'win' in sys.platform:
+        if len(split) == 5:
+            drive, path, x, y, msg = split
```

Comment:
Can you leave comments explaining what kind of situation would result in having 5 items? (Likewise for the elif checking for a length of 4) Bonus points for an example in the comment.